include/func.sh

```diff
@@ -289,3 +289,41 @@ function wait_for {
     timeout=300
     n=0; until ((n >= timeout)); do eval "$1" && break; n=$((n + 1)); sleep 1; done; ((n < timeout))
 }
+
+external_dns_annotate_uaa() {
+    local ns=$1;shift
+    local domain=$1;shift
+    kubectl annotate svc uaa-uaa-public \
+            -n "$ns" \
+            "external-dns.alpha.kubernetes.io/hostname=uaa.${domain}, *.uaa.${domain}"
```

Comment:
how will it handle parallel deployments? won't we need to use a sud domain? ` uaa.${cluster name}.${domain}`

Reply:
AFAICT the domain we see here comes from the `GKE_DNSDOMAIN`. While that has a default it can be overridden by the user. So I guess the pipeline using catapult could do that, inserting configuration data (pipeline name, diego/eirini, sa/ha), and time-dependent data. The latter would be what would allow parallel deployments.

Reply:
That's the idea, for the user/ci to pass a domain that can be used. I tested deploying clusters as `*.gke-lb.ci.kubecf.charmedquarks.me` and `gke-ingress.ci.kubecf.charmedquarks.me`, but with that specific domain did not work.
It should be possible to create a domain that works, and if not, we may need to just have a list of empty domains for grabs.